src/redux/actions/sync.js

```diff
@@ -96,56 +96,67 @@ export function doGetSync(passedPassword, callback) {
     });
 
     const data = {};
-    Lbry.sync_hash().then(hash => {
-      Lbryio.call('sync', 'get', { hash }, 'post')
-        .then(response => {
-          const syncHash = response.hash;
-          data.syncHash = syncHash;
-          data.syncData = response.data;
-          data.hasSyncedWallet = true;
-          if (response.changed) {
-            return Lbry.sync_apply({ password, data: response.data }).then(
-              ({ hash: walletHash, data: walletData }) => {
-                dispatch({ type: ACTIONS.GET_SYNC_COMPLETED, data });
-
-                if (walletHash !== syncHash) {
-                  // different local hash, need to synchronise
-                  dispatch(doSetSync(syncHash, walletHash, walletData));
-                  handleCallback();
-                }
-              }
-            );
-          }
+
+    Lbry.wallet_status()
+      .then(status => {
+        if (status.is_locked) {
+          return Lbry.wallet_unlock({ password });
+        }
+      })
+      .then(() => Lbry.sync_hash())
+      .then(hash => Lbryio.call('sync', 'get', { hash }, 'post'))
+      .then(response => {
+        const syncHash = response.hash;
+        data.syncHash = syncHash;
+        data.syncData = response.data;
+        data.hasSyncedWallet = true;
+
+        if (response.changed) {
+          return Lbry.sync_apply({ password, data: response.data });
+        }
+      })
+      .then(response => {
```

Comment:
I guess this is also the same concern as my other comment. Should the callback only be called after `sync_apply` is done? And what happens if `sync_apply` fails?

Reply:
If it fails here it will be caught down below